cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowStatsSLAHttpHandler.java

```diff
@@ -89,4 +111,181 @@ public void workflowStats(HttpRequest request, HttpResponder responder,
     }
     responder.sendJson(HttpResponseStatus.OK, workflowStatistics);
   }
+
+  @GET
+  @Path("apps/{app-id}/workflows/{workflow-id}/runs/{run-id}/statistics")
+  public void workflowRunDetail(HttpRequest request, HttpResponder responder,
+                                @PathParam("namespace-id") String namespaceId,
+                                @PathParam("app-id") String appId,
+                                @PathParam("workflow-id") String workflowId,
+                                @PathParam("run-id") String runId,
+                                @QueryParam("limit") int limit,
+                                @QueryParam("interval") String interval) throws Exception {
+    if (limit < 0) {
+      throw new BadRequestException("Count has to be greater than or equal to 0");
```

Comment:
ignore the comment. i see we use ``HttpExceptionHandler`` to handle these exceptions.